src/python/pants/reporting/reporting.py

```diff
@@ -112,6 +111,18 @@ def initialize(self, run_tracker, all_options, start_time=None):
       raise ValueError(
         "Flags zipkin-trace-id and zipkin-parent-id must both either be set or not set."
       )
+    if trace_id and (len(trace_id) != 16 and len(trace_id) != 32 or \
+      any(ch not in set('0123456789abcdefABCDEF') for ch in trace_id)):
+      raise ValueError(
+        "Value of the flag zipkin-trace-id must be a 16-character or 32-character hex string. "
+        + "Got {}.".format(trace_id)
+      )
+    if parent_id and (len(parent_id) != 16 or \
+      any(ch not in set('0123456789abcdefABCDEF') for ch in parent_id)):
```

Comment:
Maybe pull out a helper function called `is_hex_string` to avoid repeating this twice?